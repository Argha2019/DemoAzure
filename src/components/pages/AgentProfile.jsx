import React from 'react';
import Cookies from 'js-cookie';
import axios from 'axios';
import AgentTemplate from "../templates/AgentTemplate/Template.jsx";
import { Row, Col, Form, Button } from 'react-bootstrap';
import { encryptByAES, decryptByAES } from './Encryption.jsx';
import Config from "../../config/urls.jsx";

const alphabetRegex = RegExp(/^[A-Za-z]{1,20}$/);
const placeRegex = RegExp(/^[a-zA-Z ]{1,49}$/);
const mailRegex = RegExp(/^[a-zA-Z0-9.!#$%&'*+/=?^_/'{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/);

const formValid = ({ formErrors, ...rest }) => {
	let valid = true;

	// validate form errors being empty
	Object.values(formErrors).forEach(val => {
		val.length > 0 && (valid = false);
	});

	// validate the form was filled out
	Object.values(rest).forEach(val => {
		val === null && (valid = false);
	});

	return valid;
};

export default class AgentProfile extends React.Component {
	constructor(props) {
		super(props);
		this.state = { readOnly: true }
		this.handleChange = this.handleChange.bind(this);
		this.editProfile = this.editProfile.bind(this);
		this.cancelProfile = this.cancelProfile.bind(this);
		this.submitProfile = this.submitProfile.bind(this);
		let profileData = {};
		try {
			profileData = JSON.parse(decryptByAES(Cookies.get('agentData')))[0];
		}
		catch (e) {
			profileData = {
				firstName: '',
				lastName: '',
				username: '',
				role: '',
				phone: '',
				mail: '',
				addressLine1: '',
				addressLine2: '',
				state: '',
				country: '',
				pincode: '',
				dateOfBirth: '',
				status: ''
			};
		}
		this.state = {
			successMessage: '',
			errorMessage: '',
			firstName: profileData.firstName,
			lastName: profileData.lastName,
			dateOfBirth: profileData.dob,
			phone: decryptByAES(profileData.phone),
			mail: decryptByAES(profileData.mail),
			addressLine1: profileData.addressLine1,
			addressLine2: profileData.addressLine2,
			state: profileData.state,
			country: profileData.country,
			pincode: profileData.pinCode,
			role: profileData.roles,
			username: decryptByAES(profileData.username),
			status: profileData.status,
			password: profileData.password,
			edit: false,
			submit: false,
			user: {},
			formErrors: {
				firstName: '',
				lastName: '',
				username: '',
				role: '',
				phone: '',
				mail: '',
				addressLine1: '',
				addressLine2: '',
				state: '',
				country: '',
				pincode: '',
				dateOfBirth: ''
			}

		};
	}

	dateValidator() {
		var date = new Date();
		date.setDate(date.getDate() - 1);
		var dateEntered = new Date(document.getElementById('dob1').value);
		if (date.getFullYear() - dateEntered.getFullYear() < 18) {
			return true;
		}

		if (date.getFullYear() - dateEntered.getFullYear() == 18) {

			//CD: 11/06/2018 and DB: 15/07/2000. Will turned 18 on 15/07/2018.
			if (dateEntered.getMonth() > date.getMonth()) {
				return true;
			}
			if (dateEntered.getMonth() == date.getMonth()) {
				//CD: 11/06/2018 and DB: 15/06/2000. Will turned 18 on 15/06/2018.
				if (dateEntered.getDate() > date.getDate()) {
					return true;
				}
			}
		}
	}

	handleChange(e) {
		e.preventDefault();
		const { name, value } = e.target;
		let formErrors = { ...this.state.formErrors };
		switch (name) {
			case "firstName":
				formErrors.firstName =
					!(alphabetRegex.test(value)) ? "Name should contain only alphabets and character limit is (2-20) characters" : "";
				break;
			case "lastName":
				formErrors.lastName =
					!(alphabetRegex.test(value)) ? "Name should contain only alphabets and character limit is (2-20) characters" : "";
				break;
			case "phone":
				formErrors.phone =
					(value.length == 10) ? "" : "Number of digit must be 10";
				break;
			case "mail":
				formErrors.mail =
					!(mailRegex.test(value)) ? "Enter valid mail" : "";
				break;
			case "addressLine1":
				formErrors.addressLine1 =
					(value.length > 50) ? "Exceeded the characters limit of 50" : "";
				break;
			case "addressLine2":
				formErrors.addressLine2 =
					(value.length > 50) ? "Exceeded the characters limit of 50" : "";
				break;
			case "state":
				formErrors.state =
					!(placeRegex.test(value)) ? "Only alphabets allowed and character limit is 50" : "";
				break;
			case "country":
				formErrors.country =
					!(placeRegex.test(value)) ? "Only alphabets allowed and character limit is 50" : "";
				break;
			case "pincode":
				formErrors.pincode =
					(value.length === 6) ? "" : "Number of digit must be 6";
				break;
			case "dateOfBirth":
				formErrors.dateOfBirth =
					(this.dateValidator()) ? "Current or future date is not allowed. The user must be 18 years old." : "";
				break;
			default:
				break;
		}
		this.setState({ formErrors, [name]: value });
	}

	editProfile(e) {
		e.preventDefault();
		this.setState({
			edit: !this.state.edit

		})
		this.setState(prevState => ({ readOnly: !prevState.readOnly }))

	}
	cancelProfile(e) {
		e.preventDefault();
		this.setState({
			edit: !this.state.edit
		})
		this.setState(prevState => ({ readOnly: !prevState.readOnly }))

	}

	submitProfile(e) {
		e.preventDefault();
		if (formValid(this.state)) {
			let user = {
				"id": encryptByAES((JSON.parse(decryptByAES(Cookies.get('agentData')))[0].id)),
				"firstName": this.state['firstName'],
				"lastName": this.state['lastName'],
				"mail": encryptByAES(this.state['mail']),
				"username": encryptByAES(this.state['username']),
				"roles": this.state['role'],
				"phone": encryptByAES(this.state['phone']),
				"addressLine1": this.state['addressLine1'],
				"addressLine2": this.state['addressLine2'],
				"state": this.state['state'],
				"country": this.state['country'],
				"pinCode": this.state['pincode'],
				"dob": this.state['dateOfBirth'],
				"status": JSON.parse(decryptByAES(Cookies.get('agentData')))[0].status,
				"password": encryptByAES("Abcde@123"),
				"accessToken": JSON.parse(decryptByAES(Cookies.get('agentData')))[0].accessToken,
				"refreshToken": JSON.parse(decryptByAES(Cookies.get('agentData')))[0].refreshToken,
				"expireTime": JSON.parse(decryptByAES(Cookies.get('agentData')))[0].expireTime
			}

			var url = Config.serverUrl + '/signup/update';

			axios.put(url, user)
				.then(
					response => {
						this.setState({ user: response.data });
						this.setState({ submit: true });
						Cookies.set('agentData', encryptByAES(JSON.stringify([user])));

						if (response.data === true) {
							this.setState({ successMessage: "You have successfully updated the profile." });
							this.setState({errorMessage : ""});
							window.scrollTo({
								top: 0,
								behavior: 'smooth'
							  });
						}
						else {
							this.setState({ errorMessage: "Please try again." });
							window.scrollTo({
								top: 0,
								behavior: 'smooth'
							  });
						}
					}

				)

			this.setState({
				edit: !this.state.edit
			})
			this.setState(prevState => ({ readOnly: !prevState.readOnly }))
		}
		else {
			this.setState({ errorMessage: "Please fill all fields correctly." });
			window.scrollTo({
				top: 0,
				behavior: 'smooth'
			  });
		}
	}


	render() {
		const { formErrors } = this.state;

		return (
			<AgentTemplate>
				<div className="register-wrap">
					<br />
					<br />
					<br />
					<br />
					<h1 className="fontWhite">Profile</h1>
					<div className="Oms register-html">

						{this.state.successMessage.length > 0 && (
							<span className="successMessage">{this.state.successMessage}</span>
						)}

						{this.state.errorMessage.length > 0 && (
							<span className="errorMessage">{this.state.errorMessage}</span>
						)}
						<br/><br/>
						<Form className="MyForm register-form">
							<Form.Group as={Row} controlId="username">
								<Col sm={4} md={3} className="alignLabel">
									<Form.Label className="Labelling label">Username</Form.Label>
								</Col>
								<Col sm={8} md={9}>
									<Form.Control name="username" type="text" defaultValue={this.state.username} ref="username" readOnly required />
								</Col>
							</Form.Group>

							<Form.Group as={Row} controlId="role">
								<Col sm={4} md={3} className="alignLabel">
									<Form.Label className="Labelling label">Role</Form.Label>
								</Col>
								<Col sm={8} md={9}>
									<Form.Control name="role" type="text" defaultValue={this.state.role} ref="role" readOnly required />
								</Col>
							</Form.Group>

							<Form.Group as={Row} controlId="firstName">
								<Col sm={4} md={3} className="alignLabel">
									<Form.Label className="Labelling label">First Name</Form.Label>
								</Col>
								<Col sm={8} md={9}>
									<Form.Control className="firstName" name="firstName" type="text" defaultValue={this.state.firstName} ref="firstName" readOnly={!this.state.readOnly} onChange={this.handleChange} required />
									{formErrors.firstName.length > 0 && (
										<span className="errorMessage">{formErrors.firstName}</span>
									)}
								</Col>
							</Form.Group>

							<Form.Group as={Row} controlId="lastName">
								<Col sm={4} md={3} className="alignLabel">
									<Form.Label className="Labelling label">Last Name</Form.Label>
								</Col>
								<Col sm={8} md={9}>
									<Form.Control className="lastName" name="lastName" type="text" defaultValue={this.state.lastName} ref="lastName" readOnly={!this.state.readOnly} onChange={this.handleChange} required />
									{formErrors.lastName.length > 0 && (
										<span className="errorMessage">{formErrors.lastName}</span>
									)}
								</Col>
							</Form.Group>

							<Form.Group as={Row} controlId="dateOfBirth">
								<Col sm={4} md={3} className="alignLabel">
									<Form.Label className="Labelling label">Date of Birth</Form.Label>
								</Col>
								<Col sm={8} md={9}>
									<Form.Control id="dob1" className="dateOfBirth" name="dateOfBirth" type="date" defaultValue={this.state.dateOfBirth} ref="dateOfBirth" readOnly={!this.state.readOnly} onChange={this.handleChange} required />
									{formErrors.dateOfBirth.length > 0 && (
										<span className="errorMessage">{formErrors.dateOfBirth}</span>
									)}
								</Col>
							</Form.Group>

							<Form.Group as={Row} controlId="phone">
								<Col sm={4} md={3} className="alignLabel">
									<Form.Label className="Labelling label">Phone</Form.Label>
								</Col>
								<Col sm={8} md={9}>
									<Form.Control className="phone" name="phone" type="number" defaultValue={this.state.phone} ref="phone" readOnly={!this.state.readOnly} onChange={this.handleChange} required />
									{formErrors.phone.length > 0 && (
										<span className="errorMessage">{formErrors.phone}</span>
									)}
								</Col>
							</Form.Group>

							<Form.Group as={Row} controlId="mail">
								<Col sm={4} md={3} className="alignLabel">
									<Form.Label className="Labelling label">Mail ID</Form.Label>
								</Col>
								<Col sm={8} md={9}>
									<Form.Control className="mail" name="mail" type="email" defaultValue={this.state.mail} ref="mail" readOnly={!this.state.readOnly} onChange={this.handleChange} required />
									{formErrors.mail.length > 0 && (
										<span className="errorMessage">{formErrors.mail}</span>
									)}
								</Col>
							</Form.Group>

							<Form.Group as={Row} controlId="addressLine1">
								<Col sm={4} md={3} className="alignLabel">
									<Form.Label className="Labelling label">Address Line 1</Form.Label>
								</Col>
								<Col sm={8} md={9}>
									<Form.Control className="addressLine1" name="addressLine1" type="text" defaultValue={this.state.addressLine1} ref="addressLine1" readOnly={!this.state.readOnly} onChange={this.handleChange} required />
									{formErrors.addressLine1.length > 0 && (
										<span className="errorMessage">{formErrors.addressLine1}</span>
									)}
								</Col>
							</Form.Group>

							<Form.Group as={Row} controlId="addressLine2">
								<Col sm={4} md={3} className="alignLabel">
									<Form.Label className="Labelling label">Address Line 2</Form.Label>
								</Col>
								<Col sm={8} md={9}>
									<Form.Control className="addressLine2" name="addressLine2" type="text" defaultValue={this.state.addressLine2} ref="addressLine2" readOnly={!this.state.readOnly} onChange={this.handleChange} required />
									{formErrors.addressLine2.length > 0 && (
										<span className="errorMessage">{formErrors.addressLine2}</span>
									)}
								</Col>
							</Form.Group>

							<Form.Group as={Row} controlId="state">
								<Col sm={4} md={3} className="alignLabel">
									<Form.Label className="Labelling label">State</Form.Label>
								</Col>
								<Col sm={8} md={9}>
									<Form.Control className="state" name="state" type="text" defaultValue={this.state.state} ref="state" readOnly={!this.state.readOnly} onChange={this.handleChange} required />
									{formErrors.state.length > 0 && (
										<span className="errorMessage">{formErrors.state}</span>
									)}
								</Col>
							</Form.Group>

							<Form.Group as={Row} controlId="country">
								<Col sm={4} md={3} className="alignLabel">
									<Form.Label className="Labelling label">Country</Form.Label>
								</Col>
								<Col sm={8} md={9}>
									<Form.Control className="country" name="country" type="text" defaultValue={this.state.country} ref="country" readOnly={!this.state.readOnly} onChange={this.handleChange} required />
									{formErrors.country.length > 0 && (
										<span className="errorMessage">{formErrors.country}</span>
									)}
								</Col>
							</Form.Group>

							<Form.Group as={Row} controlId="pincode">
								<Col sm={4} md={3} className="alignLabel">
									<Form.Label className="Labelling label">Pincode</Form.Label>
								</Col>
								<Col sm={8} md={9}>
									<Form.Control className="pincode" name="pincode" type="number" defaultValue={this.state.pincode} ref="pincode" readOnly={!this.state.readOnly} onChange={this.handleChange} required />
									{formErrors.pincode.length > 0 && (
										<span className="errorMessage">{formErrors.pincode}</span>
									)}
								</Col>
							</Form.Group>
							{this.state.edit == false && (
								<div className="editButtonDiv">
									<Button className="RegisterButton" variant="primary" type="submit" onClick={this.editProfile}>Edit</Button>
								</div>
							)}

							{this.state.edit == true && (
								<div className="editProfileDiv">
									<Button className="RegisterButton" id="submit1" variant="primary" type="submit" onClick={this.submitProfile}>Submit</Button>
									<span></span>

									<Button className="RegisterButton" variant="primary" type="submit" onClick={this.cancelProfile}>Cancel</Button>
								</div>
							)}
						</Form>
					</div>
				</div>
			</AgentTemplate>
		);
	}
}
