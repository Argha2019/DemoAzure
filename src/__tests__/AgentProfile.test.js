import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import AgentProfile from '../components/pages/AgentProfile.jsx';


describe('Agent Profile' , () => {
    it('render component' , () => {
        const wrapper = shallow(<AgentProfile />);
        expect(toJson(wrapper)). toMatchSnapshot();
    });

    it('on change of firstName field' , () => {
        const event = {
            preventDefault: () => {},
            target : {
                name : "firstName",
                value : "Yashika",
				readOnly : false,
            }
        };
         const wrapper = shallow(<AgentProfile />);
		 wrapper.setState({
            edit: true,
			readOnly : false,

          });
         const input = wrapper.find('.firstName');

         input.simulate('change', event);
         expect(input.length).toBe(1);
         expect(wrapper.state('firstName')).toBe("Yashika");
    });

    
    it('on change of lastName field' , () => {
        const event = {
            preventDefault: () => {},
            target : {
                name : "lastName",
                value : "Gupta",
            }
        };
         const wrapper = shallow(<AgentProfile />);
		 wrapper.setState({
            edit: true,
			readOnly : false,

          });
         const input = wrapper.find('.lastName');

         input.simulate('change', event);
         expect(input.length).toBe(1);
         expect(wrapper.state('lastName')).toBe("Gupta");
    });

    it('on change of phone field' , () => {
        const event = {
            preventDefault: () => {},
            target : {
                name : "phone",
                value : "0000000000",
            }
        };
         const wrapper = shallow(<AgentProfile />);
		 wrapper.setState({
            edit: true,
			readOnly : false,

          });
         const input = wrapper.find('.phone');

         input.simulate('change', event);
         expect(input.length).toBe(1);
         expect(wrapper.state('phone')).toBe("0000000000");
    });

    it('on change of mail field' , () => {
        const event = {
            preventDefault: () => {},
            target : {
                name : "mail",
                value : "yashi@123",
            }
        };
         const wrapper = shallow(<AgentProfile />);
		 wrapper.setState({
            edit: true,
			readOnly : false,

          });
         const input = wrapper.find('.mail');

         input.simulate('change', event);
         expect(input.length).toBe(1);
         expect(wrapper.state('mail')).toBe("yashi@123");
    });

    it('on change of addressLine1 field' , () => {
        const event = {
            preventDefault: () => {},
            target : {
                name : "addressLine1",
                value : "abcd",
            }
        };
         const wrapper = shallow(<AgentProfile />);
		 wrapper.setState({
            edit: true,
			readOnly : false,

          });
         const input = wrapper.find('.addressLine1');

         input.simulate('change', event);
         expect(input.length).toBe(1);
         expect(wrapper.state('addressLine1')).toBe("abcd");
    });

    it('on change of addressLine2 field' , () => {
        const event = {
            preventDefault: () => {},
            target : {
                name : "addressLine2",
                value : "abcd",
            }
        };
         const wrapper = shallow(<AgentProfile />);
		 wrapper.setState({
            edit: true,
			readOnly : false,

          });
         const input = wrapper.find('.addressLine2');

         input.simulate('change', event);
         expect(input.length).toBe(1);
         expect(wrapper.state('addressLine2')).toBe("abcd");
    });

    it('on change of state field' , () => {
        const event = {
            preventDefault: () => {},
            target : {
                name : "state",
                value : "def",
            }
        };
         const wrapper = shallow(<AgentProfile />);
		 wrapper.setState({
            edit: true,
			readOnly : false,

          });
         const input = wrapper.find('.state');

         input.simulate('change', event);
         expect(input.length).toBe(1);
         expect(wrapper.state('state')).toBe("def");
    });

    it('on change of country field' , () => {
        const event = {
            preventDefault: () => {},
            target : {
                name : "country",
                value : "abcd",
            }
        };
         const wrapper = shallow(<AgentProfile />);
         const input = wrapper.find('.country');
         input.simulate('change', event);
         expect(input.length).toBe(1);
         expect(wrapper.state('country')).toBe("abcd");
    });

    it('on change of pincode field' , () => {
        const event = {
            preventDefault: () => {},
            target : {
                name : "pincode",
                value : "123456",
            }
        };
         const wrapper = shallow(<AgentProfile />);
         const input = wrapper.find('.pincode');

         input.simulate('change', event);
         expect(input.length).toBe(1);
         expect(wrapper.state('pincode')).toBe("123456");
    });
    it('cancelProfile is called', () => {
            const wrapper = shallow(<AgentProfile />);
           
            const event = {
              preventDefault: () => {},
            }
 
            wrapper.setState({
             edit:true
			});
            wrapper.update();
            const form = wrapper.find('.MyForm');
            wrapper.find('.RegisterButton').last().simulate('click', event);
            expect(wrapper.state('edit')).toEqual(false);      
      });

	  it('edit button on form 1', () => {
                 const wrapper = shallow(<AgentProfile />);
                 const form = wrapper.find('.MyForm');
                 const event = {
                   preventDefault: () => {}
                 }
                 wrapper.setState({
                   edit:false
                 });
                 wrapper.update();

                 const button = form.find('.RegisterButton');
                 button.simulate('click', event);
                 
                   expect(wrapper.state('edit')).toEqual(true);
                  //  expect(wrapper.state('noField')).toEqual("Please enter value in atleast one field");
      });


	  it('edit function', () => {
                 const wrapper = shallow(<AgentProfile />);
                 const form = wrapper.find('.MyForm');
                 const event = {
                   preventDefault: () => {}
                 }
                 wrapper.setState({
                   edit:true
                 });
                 wrapper.update();
                 const button = form.find('.RegisterButton').last();
                 button.simulate('click', event);                 
                   expect(wrapper.state('edit')).toEqual(false);
                  //  expect(wrapper.state('noField')).toEqual("Please enter value in atleast one field");
      });
})
