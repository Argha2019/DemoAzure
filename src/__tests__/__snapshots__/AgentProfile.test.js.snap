// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Agent Profile render component 1`] = `
<AgentTemplate>
  <div
    className="register-wrap"
  >
    <br />
    <br />
    <br />
    <br />
    <h1
      className="fontWhite"
    >
      Profile
    </h1>
    <div
      className="Oms register-html"
    >
      <br />
      <br />
      <Form
        className="MyForm register-form"
        inline={false}
      >
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="username"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Username
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              defaultValue=""
              name="username"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="role"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Role
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              name="role"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="firstName"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              First Name
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="firstName"
              defaultValue=""
              name="firstName"
              onChange={[Function]}
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="lastName"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Last Name
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="lastName"
              defaultValue=""
              name="lastName"
              onChange={[Function]}
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="dateOfBirth"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Date of Birth
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="dateOfBirth"
              id="dob1"
              name="dateOfBirth"
              onChange={[Function]}
              readOnly={true}
              required={true}
              type="date"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="phone"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Phone
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="phone"
              defaultValue=""
              name="phone"
              onChange={[Function]}
              readOnly={true}
              required={true}
              type="number"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="mail"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Mail ID
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="mail"
              defaultValue=""
              name="mail"
              onChange={[Function]}
              readOnly={true}
              required={true}
              type="email"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="addressLine1"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Address Line 1
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="addressLine1"
              defaultValue=""
              name="addressLine1"
              onChange={[Function]}
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="addressLine2"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Address Line 2
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="addressLine2"
              defaultValue=""
              name="addressLine2"
              onChange={[Function]}
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="state"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              State
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="state"
              defaultValue=""
              name="state"
              onChange={[Function]}
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="country"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Country
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="country"
              defaultValue=""
              name="country"
              onChange={[Function]}
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="pincode"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Pincode
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="pincode"
              name="pincode"
              onChange={[Function]}
              readOnly={true}
              required={true}
              type="number"
            />
          </Col>
        </FormGroup>
        <div
          className="editButtonDiv"
        >
          <Button
            active={false}
            className="RegisterButton"
            disabled={false}
            onClick={[Function]}
            type="submit"
            variant="primary"
          >
            Edit
          </Button>
        </div>
      </Form>
    </div>
  </div>
</AgentTemplate>
`;
